napari/layers/_shapes_layer/shape_util.py

```diff
@@ -3,19 +3,49 @@
 from vispy.geometry import PolygonData
 
 
+def slice_by_plane(data):
+    """Determine if shape is entirely defined within a single 2D plane
+
+    Parameters
+    ----------
+    data : (N, D) array
+        Array of vertices specifying the shape.
+
+    Returns
+    ----------
+    key : tuple of int or bool
+        If shape is 2D returns an empty tuple, otherwise return the tuple
+        of int defining the 2D plane if the shape is entirely definied
+        within a single 2D plane. Otherwise return False as the shape is
+        invalid.
+    value : (N, 2) array or False
+        Array of vertices specifying the shape in the 2D plane or False
+        if no such specification exists.
+    """
+    # TODO make which coordinate are the displayed 2D coordinates changable
+    # instead of assuming they are the last two dimensions. This feature
+    # will be implemented here after it is implemented on the main dims model.
+    # The key can then be a tuple: (slice_dims, coords)
+    if data.shape[1] == 2:
+        return (), data
+    elif np.all(data[:, :-2] == data[0, :-2]):
+        return tuple(data[0, :-2].astype('int')), data[:, -2:]
```

Comment:
use `np.int` instead

Reply:
no, just `int`, for which `np.int` is an alias.

Reply:
ok sticking with `int`